text/0000-extended-standard-library.md

```diff
@@ -0,0 +1,370 @@
+- Feature Name: Extended Standard Library
+- Start Date: 2025-05-08
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC describes an Extended Standard Library (ESL) for Rust. The ESL will consist of a set of crates providing commonly used functionality, subject to policies which promote excellent security, quality, and ease of use.
+
+# Motivation
+[motivation]: #motivation
+
+Rust programs, on average, have many dependencies. This is largely because the Rust standard library is intentionally limited in scope. As a consequence, Rust programs are currently required to pull in many third-party dependencies to get functionality that is considered table-stakes by 2025 standards.
+
+There are three issues this proposal aims to help address, all of which are related to the large number of independent entities providing crates that are widely depended on in Rust software:
+
+### 1. Impractically-sized web of trust
+
+Reliance on a high number of third-party dependencies creates a large web of trust for any non-trivial program. Given the size of this web of trust, it is practically impossible to be confident about the trustworthiness of the full dependency tree.
+
+Being a _trustworthy_ dependency means it:
+
+* Is developed and maintained by qualified, non-malicious authors
```

Comment:
What about non-malicious but unfavored changes?

This can range from
- A maintainer making the API more non-ergonic for purity purposes (highly debateable on whether this is a problem)
- A maintainer shipping pre-built binaries (we saw how that went)

Flipping the perspective on this, I know with clap I've appreciated having WG-CLI as a sounding board for things I thought might be a good idea but I was unsure how they would be received.

Reply:
> With clap's current API, I find it to be so heavily abstracted and the documentation for processing OsString paths 

While a bit off topic for this RFC, `OsString` paths "just work" which was a large reason for the clap v4 design as before where you had to know they weren't supported natively and find the right incantation to get them to work.

Reply:
They "just work" if you "just want an `OsString`". Last time I wasn't just copy-pasting a previous successful use, they were a frustrating and under-documented type-system puzzle-box if you wanted to plumb in some custom validation and get a `PathBuf` out.

Reply:
> What about non-malicious but unfavored changes?

There is already an example in the ecosystem,though not quite in this RFC's scope, https://github.com/gluon-lang/lsp-types/issues/284. Some downstream users are so unfavor the changes and start creating forks. I think it's a lose-lose for the ecosystem. The problem will be worse if the crate in question is in ESL.